rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -1053,11 +1053,9 @@ static String defaultObjectToString(Scriptable obj) {
         if (Undefined.isUndefined(obj)) return "[object Undefined]";
 
         String tag = null;
-        if (ScriptableObject.hasProperty(obj, SymbolKey.TO_STRING_TAG)) {
-            Object tagValue = ScriptableObject.getProperty(obj, SymbolKey.TO_STRING_TAG);
-            if (tagValue instanceof String) {
-                tag = (String) tagValue;
-            }
+        Object tagValue = ScriptableObject.getProperty(obj, SymbolKey.TO_STRING_TAG);
+        if (tagValue != Scriptable.NOT_FOUND && tagValue instanceof CharSequence) {
```

Comment:
cool, not_found is not a char sequence, the means you can let out this check also.
But i guess this makes to code less readable - maybe you have to add then a small comment about the missing not_found check

Reply:
Thanks for your feedback! That looks better. As suggested, I removed the explicit check for NOT_FOUND and added a related comment.